x/gamm/pool-models/stableswap/pool.go

```diff
@@ -154,37 +126,47 @@ func (p Pool) getLiquidityIndexMap() map[string]int {
 	return liquidityIndexMap
 }
 
-func reorderDecCoinSliceByDenom(coins []sdk.DecCoin, first string, second string) ([]sdk.DecCoin, error) {
-	newCoins := make([]sdk.DecCoin, len(coins))
+func (p Pool) scaledSortedPoolReserves(first string, second string, round osmomath.RoundingDirection) ([]osmomath.BigDec, error) {
+	reorderedLiquidity, reorderedScalingFactors, err := p.reorderReservesAndScalingFactors(first, second)
+	if err != nil {
+		return nil, err
+	}
+	return osmomath.DivCoinAmtsByU64ToBigDec(reorderedLiquidity, reorderedScalingFactors, round)
```

Comment:
nit: every existing caller of this function grabs first and second element from the slice returned here. What do you think about simply having 3 returns here?

We could also have a sanity error check that the first 2 elements actually exist

Reply:
Yeah, could def add it in another PR!